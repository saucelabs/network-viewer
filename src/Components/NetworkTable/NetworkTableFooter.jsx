import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import Styles from './NetworkTableFooter.styles.scss';
import { formatSize, formatTime } from './../../utils';

const context = classNames.bind(Styles);

const NetworkTableFooter = ({ dataSummary, showAllInfo }) => (
  <div className={context('footer')}>
    {showAllInfo ? (
      <>
        <span>{`${dataSummary.get('totalRequests')} requests`}</span>
        <span>{`${formatSize(dataSummary.get('totalTransferredSize'))} transferred`}</span>
        <span>{`${formatSize(dataSummary.get('totalUncompressedSize'))} resources`}</span>
        <span>{`Finished: ${formatTime(dataSummary.get('finish'))}`}</span>
        <span>{`DOMContentLoaded: ${formatTime(dataSummary.get('timings').DOMContentLoaded)}`}</span>
        <span>{`Load: ${formatTime(dataSummary.get('timings').onLoad)}`}</span>
      </>
    ) : (
      <span>{`${dataSummary.get('totalRequests')} requests`}</span>
    )}
  </div>
);

NetworkTableFooter.propTypes = {
  dataSummary: PropTypes.object.isRequired,
  showAllInfo: PropTypes.bool,
};

NetworkTableFooter.defaultProps = {
  showAllInfo: true,
};

export default NetworkTableFooter;
