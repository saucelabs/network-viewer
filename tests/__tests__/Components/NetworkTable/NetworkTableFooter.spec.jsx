import React from 'react';
import { Map } from 'immutable';
import { mount } from 'enzyme';

import NetworkTableFooter from './../../../../src/Components/NetworkTable/NetworkTableFooter';

describe('NetworkTableFooter', () => {
  it('renders without crashing', () => {
    const element = mount(
      <div>
        <NetworkTableFooter
          dataSummary={new Map({
            totalRequests: 20,
            totalTransferredSize: 2034,
            totalUncompressedSize: 1035,
            timings: {
              onLoad: 999,
              DOMContentLoaded: 2000,
            },
          })}
          showAllInfo
        />
      </div>,
    );
    expect(element).toMatchSnapshot();
  });
});
