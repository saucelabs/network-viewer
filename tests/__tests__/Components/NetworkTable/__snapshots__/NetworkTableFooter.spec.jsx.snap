// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkTableFooter renders without crashing 1`] = `
<div>
  <NetworkTableFooter
    dataSummary={
      Immutable.Map {
        "totalRequests": 20,
        "totalTransferredSize": 2034,
        "totalUncompressedSize": 1035,
        "timings": Object {
          "DOMContentLoaded": 2000,
          "onLoad": 999,
        },
      }
    }
    showAllInfo={true}
  >
    <div
      className="network-table-footer"
    >
      <span>
        20 requests
      </span>
      <span>
        2 KB transferred
      </span>
      <span>
        1 KB resources
      </span>
      <span>
        Finished: NaNm
      </span>
      <span>
        DOMContentLoaded: 2s
      </span>
      <span>
        Load: 999ms
      </span>
    </div>
  </NetworkTableFooter>
</div>
`;
